src/main/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptor.java

```diff
@@ -437,14 +546,68 @@ public FormValidation doCheckSecurityGroups(
     }
 
     @Restricted(DoNotUse.class)
+    @InjectOsAuth
+    public ComboBoxModel doFillAvailabilityZoneItems(
+            @QueryParameter String availabilityZone,
+            @QueryParameter String endPointUrl, @QueryParameter String identity, @QueryParameter String credential, @QueryParameter String zone
+    ) {
+        // Support for availabilityZones is optional in OpenStack, so this is a f:combobox not f:select field.
+        // Therefore we suggest some options if we can, but if we can't then we assume it's because they're not needed.
+        final ComboBoxModel m = new ComboBoxModel();
+        try {
+            if (haveServerDetails(endPointUrl, identity, credential, zone)) {
+                final Openstack openstack = Openstack.Factory.get(endPointUrl, identity, credential, zone);
+                for (final AvailabilityZone az : openstack.getAvailabilityZones()) {
+                    final String value = az.getZoneName();
+                    m.add(value);
+                }
+            }
+        } catch (AuthenticationException | FormValidation | ConnectionException ex) {
+            LOGGER.log(Level.FINEST, "Openstack call failed", ex);
+        } catch (Exception ex) {
+            LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+        }
+        return m;
+    }
+
+    @Restricted(DoNotUse.class)
     public FormValidation doCheckAvailabilityZone(
             @QueryParameter String value,
-            @RelativePath("../../slaveOptions") @QueryParameter("availabilityZone") String def
-    ) {
+            @RelativePath("../../slaveOptions") @QueryParameter("availabilityZone") String def,
```

Comment:
Does ` @InjectOsAuth` have a problem with this? I recollect it does with `doCheck`...